cyphers.py

```diff
@@ -41,18 +58,23 @@ def letter_frequency(c):
 elif mode == 'd':
     message1 = input('What is the message you want to decrypt? ')
     listmessage1 = list(message1)
-    freqall = []
-    for i in range(26):
-        shift = i
-        decrypt = new_message(listmessage1, encription, shift)
-        frequency = []
-        (map(letter_frequency, decrypt))
-        for i in range(len(decrypt)):
-            frequency.append(letter_frequency(decrypt[i]))
-        freqall.append(sum(frequency))
-    for x in freqall:
-        if x == max(freqall):
-            print(''.join(decrypt))
+
+    decrypted_strings = []
+    for shift in range(26):
+        # Each entry in the decrypted_strings list is itself a list
+        # of (decrypted) characters.
+        decrypted_strings.append(new_message(listmessage1, encription, shift))
+
+    best_score = 0
+    best_score_shift = 0
+
+    for shift in range(26):
```

Comment:
Now, for each entry in the list of possible decrypted strings, compute the `frequency_score` for that string.

Each step along the way, if the frequency score that we are looking at now is better (higher) than any score we have seen so far, then it is the "best" frequency score and we want to record it as such. We need to record it so that we can know whether the next or the next or the next decrypted string is "better" or not.

Also, we keep track of the "shift" value associated with the best score, so that we can actually pluck that decrypted string out of the `decrypted_strings` list at the end of the exercise. Alternatively we could take your approach like this:

```
    for x in decrypted_strings:
        if frequency_score(x) == best_score:
            print(''.join(x))
```
But that seemed clunkier to me than simply recording the list index of the best-scoring string, at the moment when we discover that it is the best-scoring string.